### PR TITLE
BZ: 2022273 - Add example for scheduling a result server pod on a worker node

### PR DESCRIPTION
**Pull request by kmccarron-rh**
https://bugzilla.redhat.com/show_bug.cgi?id=2022273

Applies to versions 4.6+

Direct link to new example: 
https://deploy-preview-38680--osdocs.netlify.app/openshift-enterprise/latest/security/compliance_operator/compliance-scans.html#running-compliance-scans-worker-node_compliance-operator-scans

### CODE REVIEW

File: modules/running-compliance-scans-worker-node.adoc [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+This is helpful for those environments where control plane nodes are not permitted to mount persistent volumes.
+
+.Procedure
+
+. Inspect the `ScanSetting` object by running:
++
+[source,terminal]
+----
+$ oc describe scansettings default -n openshift-compliance
```
Comment (jhrozek) [inline modules/running-compliance-scans-worker-node.adoc:18]: Note the name of the ScanSetting below -- it's not default, just based on the default with the minimal changes needed to run on workers. I think explaining the default SS is a good idea unless we already do it elsewhere in the doc, but the current flow gives the idea that the example describes the `scansetting/default` object wihch is not the case.

**Comment by kmccarron-rh** [comment]
@jhrozek Bump on this one. PTAL

File: modules/running-compliance-scans-worker-node.adoc [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+[id="running-compliance-scans-worker-node_{context}"]
+=  Scheduling the result server pod on a worker node
+
+The result server pod is used to mount the persistent volume (PV) storage volume that stores the raw Asset Reporting Format (ARF) scan results. The `nodeSelector` and `tolerations` attributes allow you to configure the location of the result server pod.
+
+This is helpful for those environments where control plane nodes are not permitted to mount persistent volumes.
+
+.Procedure
+
+. Inspect the `ScanSetting` object by running:
```
Comment (jhrozek) [inline modules/running-compliance-scans-worker-node.adoc:14]: One more nit -- this object doesn't exist by default. So we can't tell the user to inspect it before it's created

File: modules/running-compliance-scans-worker-node.adoc (hunk 1 above)
Comment (jhrozek) [reply modules/running-compliance-scans-worker-node.adoc:18]: Resolved

**Comment by jhrozek** [comment]
Sorry for the late reply. I included one more nit, but overall I think this is good.

**Comment by kmccarron-rh** [comment]
Reworked the example to show the steps to create the CR.  Note, the output also displays  a new strictNodeScan attribute. Is this expected?

**Review by jhrozek — APPROVED** [review]
LGTM

**Comment by jhrozek** [comment]
> Reworked the example to show the steps to create the CR. Note, the output also displays a new strictNodeScan attribute. Is this expected?

Yes, this is an unrelated addition to the operator (covered in the release notes)

**Comment by kmccarron-rh** [comment]
@pdhamdhe -PTAL  thank you!

File: modules/running-compliance-scans-worker-node.adoc [hunk 3]
```diff
@@ -0,0 +1,88 @@
+// Module included in the following assemblies:
+//
+// * security/compliance_operator/compliance-scans.adoc
+
+[id="running-compliance-scans-worker-node_{context}"]
+=  Scheduling the result server pod on a worker node
+
+The result server pod is used to mount the persistent volume (PV) storage volume that stores the raw Asset Reporting Format (ARF) scan results. The `nodeSelector` and `tolerations` attributes allow you to configure the location of the result server pod.
```
Comment (sagidlow) [inline modules/running-compliance-scans-worker-node.adoc:8]:
```suggestion
The result server pod mounts the persistent volume (PV) that stores the raw Asset Reporting Format (ARF) scan results. The `nodeSelector` and `tolerations` attributes enable you to configure the location of the result server pod.
```

Suggested for a simpler sentence and minimalism.

File: modules/running-compliance-scans-worker-node.adoc [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+
+This is helpful for those environments where control plane nodes are not permitted to mount persistent volumes.
+
+.Procedure
+. Create a `ScanSetting` custom resource (CR) for the Compliance Operator:
```
Comment (sagidlow) [inline modules/running-compliance-scans-worker-node.adoc:13]: Should there be a command at the end of this sentence? If not, then replace : with a period.

File: modules/running-compliance-scans-worker-node.adoc [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+- master
+scanTolerations:
+  - operator: Exists
+schedule: 0 1 * * *
+----
+<1> The Compliance Operator uses this node to store scan results in ARF format.
+<2> The result server pod tolerates all taints.
+
+
+.. Create the `ScanSetting` CR by running the following command:
```
Comment (sagidlow) [inline modules/running-compliance-scans-worker-node.adoc:43]:
```suggestion
.. Create the `ScanSetting` CR, run the following command:
```

> From our [contributing to docs guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#code-blocks-command-syntax-and-example-output)
If a step in a procedure is to run a command, make sure that the step text includes an explicit instruction to "run" or "enter".

File: modules/running-compliance-scans-worker-node.adoc [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+[id="running-compliance-scans-worker-node_{context}"]
+=  Scheduling the result server pod on a worker node
+
+The result server pod is used to mount the persistent volume (PV) storage volume that stores the raw Asset Reporting Format (ARF) scan results. The `nodeSelector` and `tolerations` attributes allow you to configure the location of the result server pod.
+
+This is helpful for those environments where control plane nodes are not permitted to mount persistent volumes.
+
+.Procedure
+. Create a `ScanSetting` custom resource (CR) for the Compliance Operator:
+.. Define the following `ScanSetting` CR, and save the YAML file, for example, `rs-workers.yaml`:
```
Comment (sagidlow) [inline modules/running-compliance-scans-worker-node.adoc:14]:
```suggestion
.. Define the`ScanSetting` CR, and save the YAML file, for example, `rs-workers.yaml`:
```

File: modules/running-compliance-scans-worker-node.adoc [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+
+.. Create the `ScanSetting` CR by running the following command:
++
+[source,terminal]
+----
+$ oc create -f rs-workers.yaml
+----
+
+.Verification
+. To verify that the `ScanSetting` object is created, run:
```
Comment (sagidlow) [inline modules/running-compliance-scans-worker-node.adoc:51]:
```suggestion
. Verify that the `ScanSetting` object is created, run the following command:
```

See above comment.

**Review by sagidlow — COMMENTED** [review]
Just a few comments. Let me know if you have any questions.

**Pull request merged**